### PR TITLE
Implement all or nothing transaction strategy for migrations.

### PR DESCRIPTION
**Pull request by jwage**
|      Q       |   A
|------------- | -----------
| Type         | feature
| BC Break     | yes
| Fixed issues | #594

#### Summary

Implement all or nothing transaction strategy for migrations.

### CODE REVIEW

File: lib/Doctrine/Migrations/Configuration/XML/configuration.xsd [hunk 1]
```diff
@@ -24,6 +24,7 @@
                     </xs:simpleType>
                 </xs:element>
                 <xs:element type="xs:string" name="migrations-directory" minOccurs="0" maxOccurs="1"/>
+                <xs:element type="xs:string" name="all-or-nothing" minOccurs="0" maxOccurs="1"/>
```
Comment (stof) [inline lib/Doctrine/Migrations/Configuration/XML/configuration.xsd]: shoudln't it be `xs:bool` ?

File: docs/en/reference/installation.rst [hunk 2]
```diff
@@ -70,6 +70,8 @@ file like the following:
 
         <migrations-directory>/path/to/migrations/classes/DoctrineMigrations</migrations-directory>
 
+        <all-or-nothing>1</all-or-nothing>
```
Comment (OskarStark) [inline docs/en/reference/installation.rst]: Here it is 1...

File: docs/en/reference/introduction.rst [hunk 3]
```diff
@@ -185,18 +185,87 @@ the migrations are executed in the correct order.
...
+
+            <name>Doctrine Sandbox Migrations</name>
+
+            <migrations-namespace>DoctrineMigrations</migrations-namespace>
+
+            <table name="doctrine_migration_versions" />
+
+            <migrations-directory>/path/to/migrations/classes/DoctrineMigrations</migrations-directory>
+
+            <all-or-nothing>true</all-or-nothing>
```
Comment (OskarStark) [inline docs/en/reference/introduction.rst]: ... here it is true.

Should be consistent

Comment (jwage) [reply docs/en/reference/introduction.rst]: The docs were updated and fixed at some point in the future. FYI, we take the string from the XML and pass it through this class https://github.com/doctrine/migrations/blob/master/lib/Doctrine/Migrations/Tools/BooleanStringFormatter.php here https://github.com/doctrine/migrations/blob/master/lib/Doctrine/Migrations/Configuration/XmlConfiguration.php#L89

Comment (OskarStark) [reply docs/en/reference/introduction.rst]: Thanks for the feedback

Comment (OskarStark) [reply docs/en/reference/introduction.rst]: Thanks for the feedback

File: docs/en/reference/introduction.rst [hunk 4]
```diff
@@ -185,18 +185,87 @@ the migrations are executed in the correct order.
...
 If you specify your own migration classes (like `DoctrineMigrations\NewMigration` in the previous
 example) you will need an autoloader unless all those classes begin with the prefix Version*,
 for example path/to/migrations/classes/VersionNewMigration.php.
+
+All or Nothing Migrations
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want your migrations to be all or nothing then you can configure your migrations behave that way.
+This means when you executed multiple migrations in a row, the whole migration will be wrapped
+in a single migration and if one of the migrations fails, the transaction will be rolled back.
```
[the remark this answers is not in the document]
Comment (jwage) [reply docs/en/reference/introduction.rst]: Looks like it. Can you check if it still exists in master and record an issue to fix it if so?

[the remark this answers is not in the document]
Comment (apfelbox) [reply docs/en/reference/introduction.rst]: @jwage oh, didn't realize that the PR is so old 😅 

The whole paragraph is gone in 2.0 and I couldn't find it in 1.8 either (that's where this PR landed).

So probably already fixed then. 👍 
Sorry for the noise.

**Pull request merged**